amazon-redshift-plugin/src/test/java/io/cdap/plugin/amazon/redshift/RedshiftPluginTestBase.java

```diff
@@ -0,0 +1,215 @@
+      stmt.execute("CREATE TABLE \"dbActionTest\" (x int, day varchar(10))");
+      // create a table that the action will truncate at the end of the run
+      stmt.execute("CREATE TABLE \"postActionTest\" (x int, day varchar(10))");
+
+      stmt.execute("CREATE TABLE my_table" +
+                     "(" +
+                     "\"ID\" INT NOT NULL," +
+                     "\"NAME\" VARCHAR(40) NOT NULL," +
+                     "\"SCORE\" REAL," +
+                     "\"GRADUATED\" BOOLEAN," +
+                     "\"NOT_IMPORTED\" VARCHAR(30)," +
+                     "\"SMALLINT_COL\" SMALLINT," +
+                     "\"BIG\" BIGINT," +
+                     "\"NUMERIC_COL\" NUMERIC(" + PRECISION + "," + SCALE + ")," +
+                     "\"DECIMAL_COL\" DECIMAL(" + PRECISION + "," + SCALE + ")," +
+                     "\"DOUBLE_PREC_COL\" DOUBLE PRECISION," +
+                     "\"DATE_COL\" DATE," +
+                     "\"TIME_COL\" TIME," +
+                     "\"TIMESTAMP_COL\" TIMESTAMP(3)," +
+                     "\"TEXT_COL\" TEXT," +
+                     "\"CHAR_COL\" CHAR(100)," +
+                     "\"BYTEA_COL\" BYTEA" +
+                     ")");
+      stmt.execute("CREATE TABLE \"MY_DEST_TABLE\" AS " +
+                     "SELECT * FROM my_table");
+      stmt.execute("CREATE TABLE your_table AS " +
+                     "SELECT * FROM my_table");
+    }
+  }
+
+  protected static void prepareTestData(Connection conn) throws SQLException {
+    try (
+      Statement stmt = conn.createStatement();
+      PreparedStatement pStmt1 =
+        conn.prepareStatement("INSERT INTO my_table " +
+                                "VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?," +
+                                "       ?, ?, ?, ?, ?, ?)");
+      PreparedStatement pStmt2 =
+        conn.prepareStatement("INSERT INTO your_table " +
+                                "VALUES(?, ?, ?, ?, ?, ?, ?, ?, ?, ?," +
+                                "       ?, ?, ?, ?, ?, ?)")) {
+
+      stmt.execute("insert into \"dbActionTest\" values (1, '1970-01-01')");
+      stmt.execute("insert into \"postActionTest\" values (1, '1970-01-01')");
+
+      populateData(pStmt1, pStmt2);
+    }
+  }
+
+  private static void populateData(PreparedStatement... stmts) throws SQLException {
+    // insert the same data into both tables: my_table and your_table
+    for (PreparedStatement pStmt : stmts) {
+      for (int i = 1; i <= 5; i++) {
+        String name = "user" + i;
+        pStmt.setInt(1, i);
+        pStmt.setString(2, name);
+        pStmt.setDouble(3, 123.45 + i);
+        pStmt.setBoolean(4, (i % 2 == 0));
+        pStmt.setString(5, "random" + i);
+        pStmt.setShort(6, (short) i);
+        pStmt.setLong(7, (long) i);
+        pStmt.setBigDecimal(8, new BigDecimal(123.45).add(new BigDecimal(i)));
+        pStmt.setBigDecimal(9, new BigDecimal(123.45).add(new BigDecimal(i)));
```

Comment:
Avoid initializing a BigDecimal with double values can lead to precision and rounding issues.

Example 
```java
BigDecimal bd1 = new BigDecimal(0.1);
BigDecimal bd2 = new BigDecimal(0.2);
BigDecimal bd3 = bd1.add(bd1);
        
System.out.println(bd3.equals(bd2)); // false
        
// if we used string to initlize this would be true as we expect
BigDecimal bds1 = new BigDecimal("0.1");
BigDecimal bds2 = new BigDecimal("0.2");
BigDecimal bds3 = bds1.add(bds1);
        
System.out.println(bds3.equals(bds2)); // true
```
Refer java doc for more info [here](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#BigDecimal-double-)

---

```suggestion
        pStmt.setBigDecimal(8, new BigDecimal("123.45").add(new BigDecimal(i)));
        pStmt.setBigDecimal(9, new BigDecimal("123.45").add(new BigDecimal(i)));
```